internal/sync/sync_noop.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build ebitensinglethread
+
+package sync
+
+import "sync"
+
+type Mutex struct{}
+
+func (m *Mutex) Lock()   {}
+func (m *Mutex) Unlock() {}
+
+type RWMutex struct{}
+
+func (m *RWMutex) Lock()    {}
+func (m *RWMutex) Unlock()  {}
+func (m *RWMutex) RLock()   {}
+func (m *RWMutex) RUnlock() {}
```

Comment:
BTW, how much would the performance be improved without mutex?

Reply:
I will do some performance testing before proceeding.

Reply:
OK. I recommend to focus only on the 'run-loop' part, and then consider the mutex part in another PR later. What do you think?